### PR TITLE
feat: updated README for suggested-blocks

### LINKED ISSUE #1625: Provide better documentation for suggested-blocks
🚧⚠️🚧 This issue is being saved for **Grace Hopper Open Source Day**. 🚧⚠️🚧 An event we're running to help girls and non-binary technologists make their first contributions to open source! Please see our list of [help wanted issues](https://github.com/google/blockly-samples/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+-label%3A%22ghc-osd%22) if you'd like to contribute! 

---

**Category**

 - Plugins

**Component**

suggested-blocks

**Is your feature request related to a problem? Please describe.**

The README example is slicing into a toolbox XML string. That is not a good idea. There's a comment saying it's not a good idea, but still, we shouldn't model this for people. Also, we should use JSON.

**Describe the solution you'd like**

Replace the existing [usage documentation](https://github.com/google/blockly-samples/tree/master/plugins/suggested-blocks#usage) with the following:
```js
import * as Blockly from 'blockly';
import * as SuggestedBlocks from '@blockly/suggested-blocks';

const toolbox = {
  "kind": "categoryToolbox",
  "contents": [
    {
      "kind": "category",
      "name": "My Category",
      "contents": [ /* your category contents */ ]
    },
    {
      'kind': 'category',
      'name': 'Frequently Used',
      'custom': 'MOST_USED',
      'categorystyle': 'frequently_used_category',
    },
    {
      'kind': 'category',
      'name': 'Recently Used',
      'custom': 'RECENTLY_USED',
      'categorystyle': 'recently_used_category',
    },
  ]
};

// Inject Blockly.
const workspace = Blockly.inject('blocklyDiv', {
  toolbox: toolbox,
});

// Initialize the plugin
SuggestedBlocks.init(workspace);
```

**Describe alternatives you've considered**

N/A

**Additional context**

https://google.github.io/blockly-samples/plugins/suggested-blocks/README

### PR DESCRIPTION
**Pull request by divziv**
## The basics

- [ ] I validated my changes
- [ ] (https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/google/blockly-samples/issues/1625

### Reason for Changes
- [x] Worked on the fork as part of the GHC OSD track

### CODE REVIEW

**Comment by divziv** [comment]
CLA checked.

File: plugins/suggested-blocks/README.md [hunk 1]
```diff
@@ -17,29 +17,38 @@ npm install @blockly/suggested-blocks --save
...
-// Inject Blockly.
+// Inject Blockly. (OSD)
 const workspace = Blockly.inject('blocklyDiv', {
-  toolbox: getNewCategories(),
+  toolbox: toolbox,
 });
 
 // Initialize the plugin
 SuggestedBlocks.init(workspace);
-```
```
Comment (johnnesky) [inline plugins/suggested-blocks/README.md:42]: I think these 3 backticks are important to preserve.

File: plugins/suggested-blocks/README.md [hunk 2]
```diff
@@ -17,29 +17,38 @@ npm install @blockly/suggested-blocks --save
...
-  // This type of insertion works in a pinch, but you would more so want to add
-  // the new categories wherever the rest of the categories are defined in the
-  // project (e.g. HTML, or a dedicated file for all the categories)
-  const indexToInsert = toolboxCategories.indexOf('</xml>');
-  // Insert the new categories into the existing toolbox XML string
-  return toolboxCategories.slice(0, indexToInsert) + mostUsedCategory + recentlyUsedCategory + toolboxCategories.slice(indexToInsert);
-}
-
-// Inject Blockly.
+// Inject Blockly. (OSD)
```
Comment (johnnesky) [inline plugins/suggested-blocks/README.md]: Is (OSD) relevant for the contents of this readme?

**Review by johnnesky — CHANGES_REQUESTED** [review]

**Comment by divziv** [comment]
Hey!
I just edited it again. Could you please check it and let me know?

File: plugins/suggested-blocks/README.md [hunk 3]
```diff
@@ -17,29 +17,39 @@ npm install @blockly/suggested-blocks --save
...
+    },
+    {
+      'kind': 'category',
+      'name': 'Recently Used',
+      'custom': 'RECENTLY_USED',
+      'categorystyle': 'recently_used_category',
+    },
+  ]
+};
+```
```
Comment (johnnesky) [inline plugins/suggested-blocks/README.md]: Whoops, looks like you added the backticks in the wrong place. The backticks should be after the entire code sample, the same as before. I believe they should be after "SuggestedBlocks.init(workspace);"

**Review by johnnesky — CHANGES_REQUESTED** [review]

**Comment by divziv** [comment]
Thanks for the tip! I just fixed that.

File: plugins/suggested-blocks/README.md (hunk 3 above)
Comment (rachel-fenichel) [reply plugins/suggested-blocks/README.md]: @divziv there's more [information about backticks/code fences](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) on the GitHub docs, if you're curious.

**Review by johnnesky — APPROVED** [review]

**Comment by johnnesky** [comment]
Nice work, thank you!

**Comment by divziv** [comment]
Thank you for helping out. It was a pleasure working on blockly. It was
also the first time I ever forked into an open source project, pulled a
request and committed changes. Thank you again for making this happen. I
appreciate your time and effort.

Sincerely,
Divya Hari Kumar

**Pull request merged**